test/statsd_instrumentation_test.rb

```diff
@@ -337,6 +337,48 @@ def test_statsd_count_with_tags
     ActiveMerchant::Gateway.singleton_class.statsd_remove_count :sync, 'ActiveMerchant.Gateway.sync'
   end
 
+  def test_statsd_respects_global_prefix_changes
+    StatsD.prefix = 'Foo'
+    ActiveMerchant::Gateway.singleton_class.extend StatsD::Instrument
+    ActiveMerchant::Gateway.singleton_class.statsd_count :sync, 'ActiveMerchant.Gateway.sync'
+    StatsD.prefix = 'Quc'
+
+    statsd_calls = capture_statsd_calls { ActiveMerchant::Gateway.sync }
+    assert_equal 1, statsd_calls.length
+    assert_equal "Quc.ActiveMerchant.Gateway.sync", statsd_calls.first.name
+  ensure
+    StatsD.prefix = nil
+    ActiveMerchant::Gateway.singleton_class.statsd_remove_count :sync, 'ActiveMerchant.Gateway.sync'
+  end
+
+  def test_statsd_macro_can_overwrite_prefix
+    StatsD.prefix = 'Foo'
+    ActiveMerchant::Gateway.singleton_class.extend StatsD::Instrument
+    ActiveMerchant::Gateway.singleton_class.statsd_measure :sync, 'ActiveMerchant.Gateway.sync', prefix: 'Bar'
+    StatsD.prefix = 'Quc'
+
+    statsd_calls = capture_statsd_calls { ActiveMerchant::Gateway.sync }
```

Comment:
I am not using `assert_statsd_measure` here because it automatically accounts for the prefix, which is very unhelpful in this case.